### DISCUSSION TITLE
non-volatile memory [?]

### DISCUSSION THREAD

**Discussion by spy653** [phil-opp/blog_os#886] [General]
Hi, long time fan of the blog, just wondering how far down the line / how daunting of a task would you consider non-volatile memory at this stage in the blog?

As an uninformed individual it would seem like one of the last big steps to becoming an almost proper os

**Comment by phil-opp** [comment]
Hi,

with non-volatile memory, you mean hard disks and SSDs? This isn't something that I plan to tackle soon, mainly because there are other things that seem more important to me in the short term (e.g. threads, processes, userspace, etc). But there are other resources that explain this, for example you can look at the OSDev wiki for an overview: https://wiki.osdev.org/Storage

Best,
Philipp

**Comment by ethindp** [comment]
To get this to work you'd also need to add a few blog posts explaining
things like ACPI, PCIe, and then NVMe. All three are complicated --
even with the acpi crate.

On 12/16/20, Philipp Oppermann <notifications@github.com> wrote:
> Hi,

>

> with non-volatile memory, you mean hard disks and SSDs? This isn't something
> that I plan to tackle soon, mainly because there are other things that seem
> more important to me in the short term (e.g. threads, processes, userspace,
> etc). But there are other resources that explain this, for example you can
> look at the OSDev wiki for an overview: https://wiki.osdev.org/Storage

>

> Best,

> Philipp

>

> On Sun, 6 Dec 2020, at 04:40, Spy wrote:

>>

>

>> Hi, long time fan of the blog, just wondering how far down the line / how
>> daunting of a task would you consider non-volatile memory at this stage in
>> the blog?

>

>> As an uninformed individual it would seem like one of the last big steps
>> to becoming an almost proper os

>

>

>> —

>> You are receiving this because you are subscribed to this thread.

>> Reply to this email directly, view it on GitHub
>> <https://github.com/phil-opp/blog_os/issues/886>, or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAIUGM7ZKJRTU43SMQMS4ELSTL4KTANCNFSM4UPAP7CA>.

>

>

>

> --

> You are receiving this because you are subscribed to this thread.

> Reply to this email directly or view it on GitHub:

> https://github.com/phil-opp/blog_os/issues/886#issuecomment-746131441


-- 
Signed,
Ethin D. Probst
